plugin/pkg/admission/priority/admission.go

```diff
@@ -162,6 +163,15 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 				return fmt.Errorf("failed to get default priority class: %v", err)
 			}
 		} else {
+			pcName := pod.Spec.PriorityClassName
+			// Only allow system priorities in the system namespace. This is to prevent abuse or incorrect
+			// usage of these priorities. Pods created at these priorities could preempt system critical
+			// components.
+			for _, spc := range scheduling.SystemPriorityClasses() {
```

Comment:
if priority is immutable, this should likely only be checked on create

Reply:
that's fair, i missed that this listens to both update/delete.

Reply:
`strings.HasPrefix(pcName, scheduling.SystemPriorityClassPrefix)`?

Reply:
> if priority is immutable, this should likely only be checked on create

`admitPod` is only called on create.

> strings.HasPrefix(pcName, scheduling.SystemPriorityClassPrefix)

We allow users (admins) to create other priority classes that start with SystemPriorityClassPrefix. So, we should only check for the full names of predefined priority classes.

Reply:
> We allow users (admins) to create other priority classes that start with SystemPriorityClassPrefix. So, we should only check for the full names of predefined priority classes.

In my understanding too, I agree that for security reasons that here we should only deal with exact names to avoid any unintentional leaks that might be caused if some admin creates normal namespaces with same prefix.

Reply:
> We allow users (admins) to create other priority classes that start with SystemPriorityClassPrefix. So, we should only check for the full names of predefined priority classes.

I don't think that's true:

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/scheduling/validation/validation.go#L33-L38

but that also means those priority classes wouldn't exist, and this would reject them for that reason